### ISSUE TITLE
[REGRESSION]: getByLabel change causing issues with Vuetify framework's aria-label implementation

### ISSUE THREAD

**Issue by MarkusJLechner** [microsoft/playwright#21624]
**Context:**
- GOOD Playwright Version: 1.27
- BAD Playwright Version: 1.31

The problematic change:
https://github.com/microsoft/playwright/pull/19726

**Code Snippet**
```javascript
await page.getByLabel('Nachname').fill(lastname)
```

```
Error: strict mode violation: getByRole('dialog').filter({ hasText: 'Kunde anlegen' }).getByLabel('Nachname') resolved to 2 elements:
    1) <th scope="col" aria-sort="none" role="columnheader"…>…</th> aka getByRole('columnheader', { name: 'Nachname: Nicht sortiert. Aktivieren um aufsteigend zu sortieren.' })
    2) <input type="text" id="input-148" autocomplete="off" re…/> aka getByLabel('Nachname', { exact: true })
```

```html
<th role="columnheader" scope="col" aria-label="Nachname: Nicht sortiert. Aktivieren um aufsteigend zu sortieren." aria-sort="none" class="text-start sortable">
```

**Describe the bug**
finally i found the cause of currently failing tests. vuetify framework uses aria-label for table headers. they are generated and normally not visible in the code.

the tests fail because i want to fill a input with the label `Lastname`, but because of strict, it finds the aria label lastname in the table header. 

This is not the desired behaivor.
See docs: https://playwright.dev/docs/api/class-framelocator#frame-locator-get-by-label
> Allows locating input elements by the text of the associated label.

Searching also for `aria-label` should be flagged as a option when using `getByLabel`, especially when using the `fill` function in combination.

**Comment by dgozman** [comment]
@MarkusJLechner Most likely, you should use [`exact` option](https://playwright.dev/docs/api/class-page#page-get-by-label-option-exact). Let me know if that works for you.

```ts
await page.getByLabel('Nachname', { exact: true }).fill(lastname)
```

#19726 was an intentional change, and I would like to avoid unnecessary options there.

**Comment by dgozman** [comment]
It seems like the issue has been resolved. If you still encounter problems, please file a new issue with a repro and link to this one.

**Comment by MarkusJLechner** [comment]
Missed the first notification, thanks for the clarification. As for now, I'm using more precise locators and the exact option already.
Because the tests broke after this update I wanted to mention it. Maybe it makes sense to add this behaviour also to the [docs](https://playwright.dev/docs/api/class-framelocator#frame-locator-get-by-label)? 
With the currenct explanation, it refers only to the `<label>` element. Aria is only mentioned within the `getByRole` section. I wouldn't opened this issue if i could read it there before.

**Comment by dgozman** [comment]
@MarkusJLechner Good point about the docs, let me update them!
